test/distributed/test_c10d_common.py

```diff
@@ -181,6 +182,45 @@ def test_address_already_in_use(self):
             store1 = c10d.TCPStore(addr, port, 1, True)  # noqa: F841
             store2 = c10d.TCPStore(addr, port, 1, True)  # noqa: F841
 
+    def test_multitenancy(self):
+        addr = DEFAULT_HOSTNAME
+        port = common.find_free_port()
+
+        # Use noqa to silence flake8.
+        # Need to store in an unused variable here to ensure the first
+        # object is not destroyed before the second object is created.
+        store1 = c10d.TCPStore(addr, port, 1, True, multi_tenant=True)  # type: ignore[call-arg] # noqa: F841
+        store2 = c10d.TCPStore(addr, port, 1, True, multi_tenant=True)  # type: ignore[call-arg] # noqa: F841
```

Comment:
Can we add assertions such as a set by store 1 can be accessed via a get from store2?

Reply:
Sure, I have an assertion for that in the C++ test, but I can add it here as well.